python/cudf/cudf/core/scalar.py

```diff
@@ -0,0 +1,242 @@
+# Copyright (c) 2020, NVIDIA CORPORATION.
+import numpy as np
+
+from cudf import _lib as libcudf
+from cudf.core.column.column import ColumnBase
+from cudf.core.index import Index
+from cudf.core.series import Series
+from cudf.utils.dtypes import (
+    get_allowed_combinations_for_operator,
+    to_cudf_compatible_scalar,
+)
+
+
+class Scalar(libcudf.scalar.Scalar):
+    def __init__(self, value, dtype=None):
+        """
+        A GPU-backed scalar object with NumPy scalar like properties
+        May be used in binary operations against other scalars, cuDF
+        Series, DataFrame, and Index objects. Designed to cache host
+        and device side values for the purpose of improving reuse of
+        instances across a workflow
```

Comment:
It caching device side values is an implementation detail that I don't think we should capture in the docstring here if this is a public class.